packages/@sanity/form-builder/src/inputs/files/FileInput/FileInput.tsx

```diff
@@ -259,17 +399,22 @@ export default class FileInput extends React.PureComponent<Props, FileInputState
     )
   }
 
-  // eslint-disable-next-line class-methods-use-this
-  renderSelectFileButton() {
-    // Single asset source (just a normal button)
-    // @todo add select handling here
-    return <Button mode="bleed" text="Select" />
-  }
-
   renderFields(fields: Field[]) {
     return fields.map((field) => this.renderField(field))
   }
 
+  handleSelectAssetFromSource = (assetFromSource: AssetFromSource) => {
+    const {onChange, type, resolveUploader} = this.props
+    handleSelectAssetFromSource({
+      assetFromSource,
+      onChange,
+      type,
+      resolveUploader,
+      uploadWith: this.uploadWith.bind(this),
```

Comment:
nit: instead of using .bind here you can instead change the `uploadWith` method to be a bound instance method, e.g. change it from:
```
  uploadWith(uploader: Uploader, file: File, assetDocumentProps: UploadOptions = {}) {
    //...
```
to:
```
  uploadWith = (uploader: Uploader, file: File, assetDocumentProps: UploadOptions = {}) => {
    //...
```